src/main/java/org/gridsuite/modification/server/repositories/NetworkModificationRepository.java

```diff
@@ -135,6 +138,11 @@ private void insertModifications(List<ModificationEntity> modificationsList, Lis
                         .findFirst()
                         .orElseThrow(() -> new NetworkModificationException(MOVE_MODIFICATION_ERROR));
         modificationsList.addAll(insertionIndex, modificationsToAdd);
+        int order = 0;
+        for (ModificationEntity m : modificationsList) {
+            m.setModificationsOrder(order);
+            order++;
+        }
```

Comment:
maybe 

`for (int i = 0; i < modificationsList.size(); i++) {
    modificationsList.get(i).setModificationsOrder(i);
}`